src/AbstractMCMC.jl

```diff
@@ -88,4 +88,19 @@ include("stepper.jl")
 include("transducer.jl")
 include("logdensityproblems.jl")
 
+function __init__()
+    if isdefined(Base.Experimental, :register_error_hint)
```

Comment:
A tiny performance improvement but IIRC `__init__()` can contribute to pre-compilation and loading times to a surprising extent, so in older Julia versions without error hints you can skip it completely by swapping these two lines:
 ```suggestion
function __init__()
    if isdefined(Base.Experimental, :register_error_hint)
```
```suggestion
if isdefined(Base.Experimental, :register_error_hint)
    function __init__()
```